src/rust/ide/src/controller/module.rs

```diff
@@ -51,6 +62,28 @@ pub struct Metadata {
 
 impl parser::api::Metadata for Metadata {}
 
+/// Metadata that belongs to ide.
+#[derive(Debug,Clone,Default,Deserialize,Serialize)]
+pub struct IdeMetadata {
+    /// Metadata that belongs to nodes.
+    node : HashMap<ast::ID,NodeMetadata>
+}
+
+/// Metadata of specific node.
+#[derive(Debug,Clone,Copy,Default,Serialize,Deserialize)]
+pub struct NodeMetadata {
+    /// Position in x,y coordinates.
+    pub position: Option<Position>
+}
+
+/// Used for storing node position.
+#[derive(Clone,Copy,Debug,PartialEq,Serialize,Deserialize)]
+pub struct Position {
+    /// Vector storing coordinates of the visual position.
+    pub vector:Vector2<f32>
+}
```

Comment:
This struct looks like a little bit more complicated that it needs to be. I would just define position as a Vector2<f32>, or, if you want to have a separate type for it, I would make it as a unnamed field. possibly with Shrikwrap to have direct access to x and y.
```suggestion
pub struct Position(Vector2<f32>);
```

Reply:
This is exact revert of the change previously requested by @wdanilo .
(personally, I'm fine with newtype-like single tuples, when the name cannot really introduce new information)

Reply:
@mwu-tow Explained precisely why this design was chosen.